src/basic/process-util.c

```diff
@@ -1831,11 +1834,16 @@ int get_oom_score_adjust(int *ret) {
 
         delete_trailing_chars(t, WHITESPACE);
 
-        assert_se(safe_atoi(t, &a) >= 0);
-        assert_se(oom_score_adjust_is_valid(a));
+        r = safe_atoi(t, &a);
+        if (r < 0)
+                return r;
+
+        if (!oom_score_adjust_is_valid(a))
+                return -ENODATA;
```

Comment:
`-EINVAL` is better?

Reply:
well, my thinking was that EINVAL would indicate the caller gave us some wrong parameters, a bogus request or so. But here everything the caller did was right, but the kernel returned some bogus stuff. Hence EINVAL seems wrong there.

I pondered whether EIO or ENODATA would be better here. But settled for the latter, since I think it's a bit more precise, as there's no valid data here.